src/reducers/stagedPostsByQuery.js

```diff
@@ -0,0 +1,22 @@
+import { map, get, omit } from 'lodash'
+import { appendUniq } from './util'
+
+import {
+  FETCH_POSTS,
+  MERGE_STAGED_POSTS
+} from '../actions'
+
+export default function (state = {}, action) {
+  if (action.error) return state
+  if (!get(action, 'meta.staged')) return state
+
+  let { type, payload, meta } = action
+  switch (type) {
+    case FETCH_POSTS:
+      return appendUniq(state, meta.cacheId, map(payload.posts, 'id'))
```

Comment:
I think this should be just replacing `state[meta.cacheId]` instead of appending, no? e.g. so that we don't get duplicates if the staged fetch runs multiple times